### ISSUE TITLE
Position of Config.py

### ISSUE THREAD

**Issue by MarvinTeichmann** [TensorVision/TensorVision#6]
Commit 0266779a4ad248aaa97afd476f168d1237b01b09 puts `config.py` in the top project dir. I do like that position, however the installation script is not able to handle this. 

fe9417641232aa676ed234a3e6ce51fa49667552 therefore moves `config.py` to `tensorvision/config.py` and creates a symlink. This works but is not the most beautiful solution. Do we have any other options?

**Comment by MartinThoma** [comment]
I would expect this to be a configuration file in my home directory (e.g. `~/.tensorvisionrc`). When we don't find such a configuration file, we can automatically create it with default values. We can also add a note when executing our scripts (e.g. `tv-train`) that this was used.
- Theano uses an .rc file for configuration ([source](http://deeplearning.net/software/theano/library/config.html))

**Comment by MartinThoma** [comment]
http://12factor.net/config makes a strong case for using environment variables. The advantage is that they don't get accidentally checked in.

**Comment by MarvinTeichmann** [comment]
Both solutions sound good. Do you see a way to allow the user to run configuration specific python code? 

I am considering the case of a multi-gpu cluster where the User wants to include some controlling code before training starts. I think this is not to uncommon in research.

**Comment by MartinThoma** [comment]
> Do you see a way to allow the user to run configuration specific python code?

I don't know what you mean. Could you please explain that? Probably my latest pull request answers it?

https://doc.itc.rwth-aachen.de/display/CC/Limitations+within+the+GPU+cluster uses environment variables in a GPU cluster.

**Comment by MartinThoma** [comment]
I think we should get rid of `config.py` and convert it to environment variables.

https://github.com/TensorVision/TensorVision/blob/master/tensorvision/config.py is currently

``` python
default_name = "default"
data_dir  = "DATA"
model_dir = "output"
default_config = "examples/cifar10_minimal.json"
```

I think we should use the following environment variables with the same defaults:
- `TV_DEFAULT_NAME`
- `TV_DATA_DIR`
- `TV_MODEL_DIR`
- `TV_DEFAULT_CONFIG_PATH`

**Comment by MarvinTeichmann** [comment]
> I think we should get rid of config.py and convert it to environment variables.

I do agree with that.

> I don't know what you mean. Could you please explain that? Probably my latest pull request answers it?

No it does not. The nice think about the config.py is, that user can run environment specific python Code before the training starts. This did turn out to be very nice when running on clusters. [For example](https://github.com/MarvinTeichmann/KittiSeg/blob/master/config.py) create a flag which specifies the gpu to run on. In addition this flag can be made non-optional, avoiding accidentally runs on a wrong gpu.

Sure it is possible to "hard" code this use-case, adding an env-variable which forces the user to specify a gpu. But this is not the point, what I do like is the flexibility python configuration code is offering.  

> https://doc.itc.rwth-aachen.de/display/CC/Limitations+within+the+GPU+cluster uses environment variables in a GPU cluster.

I do not see how this changes the point. Firstly the user might want to (automatically) set those env variables inside TV. (So he can run multiple trainings without much effort). Additionally the computation infrastructure might differ between Institutes. Configuration code makes it easier for users to adapt to situation we did not anticipate.

**Comment by MartinThoma** [comment]
> No it does not. The nice think about the config.py is, that user can run environment specific python Code before the training starts. This did turn out to be very nice when running on clusters. For example create a flag which specifies the gpu to run on. In addition this flag can be made non-optional, avoiding accidentally runs on a wrong gpu.

Hm. It seems to me that this is more a "plugin" for TensorVision than it is a configuration file. We should probably simply allow plugins? So, for example, we could add an environment variable `TV_PLUGIN_DIR`. All Python files in that directory will be executed before the other TV stuff executes?

**Comment by MarvinTeichmann** [comment]
This sounds like a useful solution.

**Comment by MarvinTeichmann** [comment]
Solved by #24
